rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: boolean
+        # TODO - Not empty exception, No such namespace exception
+        '401':
+          description: Unauthorized
+components:
+  schemas:
+    CommitTableRequest:
+      type: object
+      properties:
+        tableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        metadataJson:
+          type: string
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+      properties:
+        namespace:
+          type: array
+          items:
+            type: string
+          nullable: false  # Note this is ignored
+        name:
+          pattern: \S
+          type: string
+          nullable: false
+    CommitTableResponse:
+      type: object
+      properties:
+        metadataLocation:
+          type: string
+        metadataJson:
+          type: string
+    RemovePropertiesRequest:
+      type: object
+      properties:
+        namespace:
+          type: string
+        properties:
+          uniqueItems: true
+          type: array
+          items:
+            type: string
+    Catalog:
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+          description: Unique identifier for this catalog
+        name:
+          type: string
+        location:
+          type: string
+          description: Warehouse location for this catalog or URI of metastore or other identifying location
+        properties:
+          type: object
+          description: Additional catalog level properties
+          default: {}
+      required:
+        - name
+        - properties
+    CreateNamespaceRequest:
+      type: object
+      properties:
+        namespace:
+          type: array
+          description: individual levels of the namespace
+          items:
+            type: string
+        properties:
+          type: object
+          description: Configuration properties for the namespace
+          example: '{ owner: "Hank Bendickson" }'
+    RenameTableRequest:
+      type: object
+      properties:
+        sourceTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        destinationTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+    CreateTableRequest:
+      type: object
+      properties:
```

Comment:
Can you also request a specific `location`? That seems reasonable to me.

Reply:
Is this meant for `UpdateProperties` request or is this a zombie comment coming from somewhere else?

I ask because at the time you wrote this, `UpdatePropertiesRequest` didn't exist and there's been some large reordering and github has had some trouble keeping up.

Reply:
Seeing it again, I can tell it was at one point related to CreateTableRequest.  A bug in the GitHub UI.

But we’re mostly sticking to Namspaces for now.